pandas-stubs/core/groupby/generic.pyi

```diff
@@ -41,13 +43,13 @@ class NamedAgg(NamedTuple):
 
 def generate_property(name: str, klass: type[FrameOrSeries]): ...
 
-class _SeriesGroupByScalar(SeriesGroupBy):
+class _SeriesGroupByScalar(SeriesGroupBy[S1]):
     def __iter__(self) -> Iterator[tuple[Scalar, Series]]: ...
 
-class _SeriesGroupByNonScalar(SeriesGroupBy):
+class _SeriesGroupByNonScalar(SeriesGroupBy[S1]):
     def __iter__(self) -> Iterator[tuple[tuple, Series]]: ...
```

Comment:
If you're going to make `_SeriesGroupByScalar` and `_SeriesGroupByNonScalar` depend on `S1`, I _think_ (but not 100% sure) that you should change the references in `series.pyi` to be  `_SeriesGroupByScalar[S1]` and `_SeriesGroupByNonScalar[S1]`, which may require making these two classes `Generic` as well.